src/dwc_mapping.Rmd

```diff
@@ -263,84 +347,125 @@ Save to CSV:
 write_csv(taxon, here("data", "processed", "taxon.csv"), na = "")
 ```
 
-## Literature reference extension
 
-TODO: this looks like the best way to capture the checklists that contributed to the information unified here. Note, to know actual contribution, it is best to join `distributions`, `descriptions` and `speciesProfiles` and get `sourceChecklist` rather than using `sourceChecklists` in `taxa`.
-
-### Pre-processing
+```{r}
 
-...
+```
 
-Map the data to [Literature References](http://rs.gbif.org/extension/gbif/1.0/references.xml).
 
-### Term mapping
+# Create source citation function
 
-#### identifier
+All records in the extensions will be documented with a source, i.e. the link to the specific checklist taxon that was used for the specific piece of information. We build this function here first.
 
 ```{r}
-# TODO: DOI of sourceChecklist (is internal identifier)
+build_source_citation <- function(extension){
+  
+  # Join the extension and source_citation by their datasetKey:
+  
+  extension <-
+    extension %>% 
+    left_join(
+      select(checklists, datasetKey, citation),
+      by = setNames("datasetKey", "input_datasetKey")
+    ) 
+  
+  # Generate the field `source` by combining `taxonKey`, `scientificName` and `citation`. This is an example of how the final source should look like: 
+  # https://www.gbif.org/species/141264581: Nymphaea marliacea Marliac in Verloove F, Groom Q, Brosens D, Desmet P, Reyserhove L (2018). Manual of the Alien Plants of Belgium. Version 1.7. Botanic Garden Meise. Checklist dataset https://doi.org/10.15468/wtda1m
+  
+    extension %>% 
+    mutate(source = paste0(
+      "https://www.gbif.org/species/",
+      input_taxonKey,
+      ": ",
+      input_scientificName,
+      "in  ",
```

Comment:
I meant a space before and after 😄

Reply:
:smile: I thought it was a bit odd but in these things, I just trust your opinion ;-)